jvb/src/test/kotlin/org/jitsi/videobridge/cc/av1/Av1DDAdaptiveSourceProjectionTest.kt

```diff
@@ -0,0 +1,1607 @@
+/*
+ * Copyright @ 2019 - Present, 8x8 Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.cc.av1
+
+import jakarta.xml.bind.DatatypeConverter.parseHexBinary
+import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.RtpLayerDesc
+import org.jitsi.nlj.rtp.codec.av1.Av1DDPacket
+import org.jitsi.nlj.rtp.codec.av1.Av1DDRtpLayerDesc.Companion.getIndex
+import org.jitsi.nlj.util.Rfc3711IndexTracker
+import org.jitsi.rtp.rtcp.RtcpSrPacket
+import org.jitsi.rtp.rtcp.RtcpSrPacketBuilder
+import org.jitsi.rtp.rtp.RtpPacket
+import org.jitsi.rtp.rtp.header_extensions.Av1DependencyDescriptorHeaderExtension
+import org.jitsi.rtp.rtp.header_extensions.Av1DependencyDescriptorReader
+import org.jitsi.rtp.rtp.header_extensions.Av1TemplateDependencyStructure
+import org.jitsi.rtp.rtp.header_extensions.FrameInfo
+import org.jitsi.rtp.util.RtpUtils
+import org.jitsi.rtp.util.isNewerThan
+import org.jitsi.rtp.util.isOlderThan
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.LoggerImpl
+import org.jitsi.videobridge.cc.RtpState
+import org.junit.Assert
+import org.junit.Test
+import java.time.Duration
+import java.time.Instant
+import java.util.*
+import javax.xml.bind.DatatypeConverter
+
+class Av1DDAdaptiveSourceProjectionTest {
```

Comment:
This looks very neat and thorough! Too much so to read in detail :)

Reply:
It's mostly copied from the VP9 equivalent, with minor changes.